fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/base/FullItemStorage.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.fluid.base;
+
+import net.minecraft.item.Item;
+
+import net.fabricmc.fabric.api.transfer.v1.context.ContainerItemContext;
+import net.fabricmc.fabric.api.transfer.v1.fluid.FluidVariant;
+import net.fabricmc.fabric.api.transfer.v1.item.ItemVariant;
+import net.fabricmc.fabric.api.transfer.v1.storage.StoragePreconditions;
+import net.fabricmc.fabric.api.transfer.v1.storage.base.ExtractionOnlyStorage;
+import net.fabricmc.fabric.api.transfer.v1.storage.base.SingleSlotStorage;
+import net.fabricmc.fabric.api.transfer.v1.transaction.TransactionContext;
+
+/**
+ * Base implementation of a storage for a full item.
+ */
+// TODO: more javadoc
+public final class FullItemStorage implements ExtractionOnlyStorage<FluidVariant>, SingleSlotStorage<FluidVariant> {
```

Comment:
We should discuss an elegant solution to requerying. Consider that after any operation on an item, it's Storage may become invalid, so we need a good way to auto-update. I think this is within the capabilities of the current API but I'm not sure.

Reply:
This is already handled by the API, will require good javadoc to make sure nobody forgets about it though:
```java
		// Can't insert if the item is not emptyItem anymore.
		if (!context.getItemVariant().isOf(emptyItem)) return 0;
```

Reply:
There should be a utility class to do this imo, something like:
```java
Storage<FluidKey> firstKey = ItemAPIs.firstKey(myStorage);
Storage<FluidKey> firstKey = ItemAPIs.combineAll(myStorage);
```